cli_tools/gce_onestep_image_import/onestep_importer/importer.go

```diff
@@ -0,0 +1,356 @@
+		ExportedAMIPath:    args.AWSExportedAMIPath,
+		GcsComputeEndpoint: args.ComputeEndpoint,
+		GcsProject:         args.Project,
+		GcsZone:            args.Zone,
+		GcsRegion:          args.Region,
+		GcsScratchBucket:   args.ScratchBucketGcsPath,
+		GcsStorageLocation: args.StorageLocation,
+		ImageID:            args.AWSImageID,
+		Region:             args.AWSRegion,
+		ResumeExportedAMI:  args.AWSResumeExportedAMI,
+		SecretAccessKey:    args.AWSSecretAccessKey,
+		SessionToken:       args.AWSSessionToken,
+	}
+}
+
+// Run performs onestep image import.
+func Run(args *ImportArguments) (service.Loggable, error) {
+	// 1. Validate required flags that are not cloud-provider specific.
+	if err := args.validate(); err != nil {
+		return nil, err
+	}
+
+	if args.CloudProvider == "aws" {
+		return importAMI(args)
+	}
+
+	return nil, fmt.Errorf("import from cloud provider %v is currently not supported", args.CloudProvider)
+}
+
+// importAMI imports image from AWS.
+func importAMI(args *ImportArguments) (service.Loggable, error) {
+	importer, err := awsImporter.NewImporter(args.Oauth, args.buildAWSImportArgs())
+	if err != nil {
+		return nil, err
+	}
+	exportedGCSPath, err := importer.Run()
+	if err != nil {
+		return nil, err
+	}
+
+	err = runImageImport(exportedGCSPath, args)
+	if err != nil {
+		log.Println("Failed to import image.",
+			fmt.Sprintf("The image file has been copied to Google Cloud Storage, located at %v.", exportedGCSPath),
+			"To resume the import process, please directly use image import from GCS.")
+	}
+	return nil, err
+}
+
+// runImageImport imports image from the provided GCS path.
+func runImageImport(exportedGCSPath string, args *ImportArguments) error {
+	if args.Labels == nil {
+		args.Labels = make(map[string]string)
+		args.Labels["onestep-image-import"] = args.CloudProvider
+	}
+	err := onestepUtils.RunCmd("gce_vm_image_import", []string{
+		fmt.Sprintf("-image_name=%v", args.ImageName),
+		fmt.Sprintf("-client_id=%v", args.ClientID),
+		fmt.Sprintf("-os=%v", args.OS),
+		fmt.Sprintf("-source_file=%v", exportedGCSPath),
+		fmt.Sprintf("-no_guest_environment=%v", args.NoGuestEnvironment),
+		fmt.Sprintf("-family=%v", args.Family),
+		fmt.Sprintf("-description=%v", args.Description),
+		fmt.Sprintf("-network=%v", args.Network),
+		fmt.Sprintf("-subnet=%v", args.Subnet),
+		fmt.Sprintf("-zone=%v", args.Zone),
+		fmt.Sprintf("-timeout=%v", args.Timeout),
+		fmt.Sprintf("-project=%v", args.Project),
+		fmt.Sprintf("-scratch_bucket_gcs_path=%v", args.ScratchBucketGcsPath),
+		fmt.Sprintf("-oauth=%v", args.Oauth),
+		fmt.Sprintf("-compute_endpoint_override=%v", args.ComputeEndpoint),
+		fmt.Sprintf("-disable_gcs_logging=%v", args.GcsLogsDisabled),
+		fmt.Sprintf("-disable_cloud_logging=%v", args.CloudLogsDisabled),
+		fmt.Sprintf("-disable_stdout_logging=%v", args.StdoutLogsDisabled),
+		fmt.Sprintf("-no_external_ip=%v", args.NoExternalIP),
+		fmt.Sprintf("-labels=%v", keyValueString(args.Labels).String()),
```

Comment:
label += "import-from-aws"

Reply:
I added this in after the flags are parsed